src/command.cpp

```diff
@@ -360,6 +362,85 @@ static const Command _command_proc_table[] = {
 	DEF_CMD(CmdOpenCloseAirport,                               0, CMDT_ROUTE_MANAGEMENT      ), // CMD_OPEN_CLOSE_AIRPORT
 };
 
+
+/**
+ * List of flags for a command log entry
+ */
+enum CommandLogEntryFlagEnum {
+	CLEF_NONE                = 0x00, ///< no flag is set
+	CLEF_CMD_FAILED          = 0x01, ///< command failed
+	CLEF_GENERATING_WORLD    = 0x02, ///< generating world
+	CLEF_TEXT                = 0x04, ///< have command text
+	CLEF_ESTIMATE_ONLY       = 0x08, ///< estimate only
+	CLEF_ONLY_SENDING        = 0x10, ///< only sending
+	CLEF_MY_CMD              = 0x20, ///< locally generated command
+};
+DECLARE_ENUM_AS_BIT_SET(CommandLogEntryFlagEnum)
+typedef SimpleTinyEnumT<CommandLogEntryFlagEnum, byte> CommandLogEntryFlag;
+
+struct CommandLogEntry {
+	TileIndex tile;
+	uint32 p1;
+	uint32 p2;
+	uint32 cmd;
+	Date date;
+	DateFract date_fract;
+	CompanyByte current_company;
+	CompanyByte local_company;
+	CommandLogEntryFlag log_flags;
+
+	CommandLogEntry() { }
+
+	CommandLogEntry(const CommandCost &res, TileIndex tile, uint32 p1, uint32 p2, uint32 cmd, CommandLogEntryFlag log_flags)
+			: tile(tile), p1(p1), p2(p2), cmd(cmd), date(_date), date_fract(_date_fract),
+			current_company(_current_company), local_company(_local_company), log_flags(log_flags)
+	{
+		if (res.Failed()) this->log_flags |= CLEF_CMD_FAILED;
+		if (_generating_world) this->log_flags |= CLEF_GENERATING_WORLD;
```

Comment:
Wouldn't it be more clear (possibly less error prune) to set all the flags in one place? Either here (and add 4 parameters) or lower? It feels a bit weird now, with this split.
Not sure, mainly asking. I noticed others also asked about this, and it confused me too :D